lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/LocalAccessControlModule.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import io.airlift.log.Logger;
+import io.trino.plugin.base.CatalogName;
+import io.trino.spi.connector.ConnectorAccessControl;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+
+import static com.google.common.base.Suppliers.memoizeWithExpiration;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static io.trino.plugin.base.util.JsonUtils.parseJson;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class LocalAccessControlModule
+        implements Module
+{
+    private static final Logger log = Logger.get(LocalAccessControlModule.class);
+
+    @Override
+    public void configure(Binder binder)
+    {
+        configBinder(binder).bindConfig(FileBasedAccessControlConfig.class);
+    }
+
+    @Inject
+    @Provides
+    public ConnectorAccessControl getConnectorAccessControl(CatalogName catalogName, FileBasedAccessControlConfig config)
+    {
+        File configFile = new File(config.getConfigFile());
+        if (config.getRefreshPeriod() != null) {
+            return ForwardingConnectorAccessControl.of(memoizeWithExpiration(
+                    () -> {
+                        log.info("Refreshing access control for catalog '%s' from: %s", catalogName, configFile);
+                        AccessControlRules rules = parseJson(readFile(configFile), config.getJsonPointer(), AccessControlRules.class);
+                        return new FileBasedAccessControl(catalogName, rules);
+                    },
+                    config.getRefreshPeriod().toMillis(),
+                    MILLISECONDS));
+        }
+
+        AccessControlRules rules = parseJson(readFile(configFile), config.getJsonPointer(), AccessControlRules.class);
+        return new FileBasedAccessControl(catalogName, rules);
+    }
```

Comment:
Can we have this as a part of `FileBasedAccessControlModule` - so that we could use them both for `File` based and `Uri` based implementation

Reply:
refactored